src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -0,0 +1,358 @@
+            return ((OLECMDF)cmds[0].cmdf).HasFlag(OLECMDF.OLECMDF_ENABLED);
+        }
+
+        private void ShowGoldBar()
+        {
+            AssertIsForeground();
+
+            // If the gold bar is already open, do not show
+            if (_infoBarOpen)
+            {
+                return;
+            }
+
+            _infoBarOpen = true;
+
+            string message;
+            if (_experimentationService.IsExperimentEnabled(InternalFlightName))
+            {
+                message = ServicesVSResources.We_noticed_you_suspended_ReSharper_Ultimate_Restore_Visual_Studio_keybindings_to_continue_to_navigate_and_refactor;
+            }
+            else if (_experimentationService.IsExperimentEnabled(ExternalFlightName))
+            {
+                message = ServicesVSResources.We_noticed_your_keybindings_are_broken;
+            }
+            else
+            {
+                // Should never have gotten to checking this if one of the flights isn't enabled.
+                throw ExceptionUtilities.Unreachable;
+            }
+
+            KeybindingsResetLogger.Log("InfoBarShown");
+            var infoBarService = _workspace.Services.GetRequiredService<IInfoBarService>();
+            infoBarService.ShowInfoBarInGlobalView(
+                message,
+                new InfoBarUI(title: ServicesVSResources.Restore_Visual_Studio_keybindings,
+                              kind: InfoBarUI.UIKind.Button,
+                              action: RestoreVsKeybindings),
+                new InfoBarUI(title: ServicesVSResources.Use_Keybindings_for_extensions,
+                              kind: InfoBarUI.UIKind.Button,
+                              action: OpenExtensionsHyperlink),
+                new InfoBarUI(title: ServicesVSResources.Never_show_this_again,
+                              kind: InfoBarUI.UIKind.HyperLink,
+                              action: NeverShowAgain),
+                new InfoBarUI(title: "", kind: InfoBarUI.UIKind.Close,
+                              action: InfoBarClose));
+        }
+
+        private void RestoreVsKeybindings()
+        {
+            AssertIsForeground();
+
+            if (_uiShell == null)
+            {
+                _uiShell = _serviceProvider.GetService(typeof(SVsUIShell)) as IVsUIShell;
+            }
+
+            ErrorHandler.ThrowOnFailure(_uiShell.PostExecCommand(
+                    VSConstants.GUID_VSStandardCommandSet97,
+                    (uint)VSConstants.VSStd97CmdID.CustomizeKeyboard,
+                    (uint)OLECMDEXECOPT.OLECMDEXECOPT_DODEFAULT,
+                    null));
+
+            KeybindingsResetLogger.Log("KeybindingsReset");
+
+            _workspace.Options = _workspace.Options.WithChangedOption(KeybindingResetOptions.NeedsReset, false);
+        }
+
+        private void OpenExtensionsHyperlink()
+        {
+            ThisCanBeCalledOnAnyThread();
+            Process.Start(KeybindingsFwLink);
```

Comment:
This is how other experiments did it when I did the VSIX gold bar, so I copied the same technique.

---
In reply to: [155659705](https://github.com/dotnet/roslyn/pull/23605#discussion_r155659705) [](ancestors = 155659705)

Reply:
we have this
http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/Utilities/BrowserHelper.cs,57

Reply:
you mean BrowserHelper.StartBrowser ? it will use VsShellUtilities.OpenSystemBrowser(uri.AbsoluteUri) which uses VS options to determine whether it opens up embedded browser or external browser and etc.

Reply:
Missed that method.